Import Library

In [8]:
import pandas as pd
from tabulate import tabulate

Menampilkan dataset

In [9]:
dF = pd.read_csv('Corona_NLP_test.csv')

In [10]:
dF.head()

UserName  ScreenName             Location     TweetAt  \
0         1       44953                  NYC  02-03-2020   
1         2       44954          Seattle, WA  02-03-2020   
2         3       44955                  NaN  02-03-2020   
3         4       44956          Chicagoland  02-03-2020   
4         5       44957  Melbourne, Victoria  03-03-2020   

                                       OriginalTweet           Sentiment  
0  TRENDING: New Yorkers encounter empty supermar...  Extremely Negative  
1  When I couldn't find hand sanitizer at Fred Me...            Positive  
2  Find out how you can protect yourself and love...  Extremely Positive  
3  #Panic buying hits #NewYork City as anxious sh...            Negative  
4  #toiletpaper #dunnypaper #coronavirus #coronav...             Neutral

44957 rows x 6 columns

Menampilkan rata-rata, standar deviasi, nilai minimum, nilai maksimum, Q1, Q2, dan Q3 dari attribute bertipe angka.

In [11]:
dF.describe()

UserName    ScreenName
count  3798.000000   3798.000000
mean   1899.500000  46851.500000
std    1096.532489   1096.532489
min       1.000000  44953.000000
25%     950.250000  45902.250000
50%    1899.500000  46851.500000
75%    2848.750000  47800.750000
max    3798.000000  48750.000000

Menampilkan 10 record pertama dari dataset dengan attribute bertipe angka.

In [12]:
attribute_angka = dF.select_dtypes (include=['int64', 'float64'])
attribute_angka.head(9)

UserName  ScreenName
0         1       44953
1         2       44954
2         3       44955
3         4       44956
4         5       44957
5         6       44958
6         7       44959
7         8       44960
8         9       44961

Menampilkan informasi jumlah record, jumlah attribute, nama attribute, jumlah attribute dengan tipe angka.

In [13]:
# Jumlah Record
jumlah_record = len(dF)
# Jumlah atribut tipe object
objek = len(dF.dtypes[dF.dtypes == 'object'])
# Nama Attribute
atr = list(dF.columns)
atr = "\n".join(atr)
# Jumlah atribut tipe angka
angka = len(dF.dtypes[dF.dtypes == 'int64'])
dictionary = {
    "Jumlah Record" : [jumlah_record],
    "Jumlah Attribute Tipe Object" : [objek],
    "Nama Attribute":[atr],
    "Jumlah Attribute Tipe Angka" : [angka]
}
tabel = pd.DataFrame(dictionary)
print(tabulate(tabel, headers='keys', tablefmt='psql', showindex=False))

+-----------------+--------------------------------+------------------+-------------------------------+
|   Jumlah Record |   Jumlah Attribute Tipe Object | Nama Attribute   |   Jumlah Attribute Tipe Angka |
|-----------------+--------------------------------+------------------+-------------------------------|
|            3798 |                              4 | UserName         |                             2 |
|                 |                                | ScreenName       |                               |
|                 |                                | Location         |                               |
|                 |                                | TweetAt          |                               |
|                 |                                | OriginalTweet    |                               |
|                 |                                | Sentiment        |                               |
+-----------------+--------------------------------+------------

Menampilkan nilai dataset pada baris dimulai dengan angka 102 hingga data terakhir dan kolom ke 4 hingga kolom terakhir menggunakan ke 2 metode slice.

Menggunakan metode .iloc

In [19]:
# Menggunakan metode .iloc (indeks berbasis posisi)
subset_iloc = dF.iloc[102:, 3:]  # Mulai dari baris 102 dan dari kolom ke-4

# Menampilkan hasil
print("Menggunakan metode .iloc:")
print(subset_iloc)

Menggunakan metode .iloc:
         TweetAt                                      OriginalTweet  \
102   10-03-2020  Missouri AG sues televangelist for advertising...   
103   10-03-2020  President Trump: "It will go away.  Just stay ...   
104   10-03-2020  You actually need to stock pile on food first ...   
105   10-03-2020  We (@hamiltonpoverty) are calling on Canadian ...   
106   10-03-2020  Solid advice for seniors, their family members...   
...          ...                                                ...   
3793  16-03-2020  Meanwhile In A Supermarket in Israel -- People...   
3794  16-03-2020  Did you panic buy a lot of non-perishable item...   
3795  16-03-2020  Asst Prof of Economics @cconces was on @NBCPhi...   
3796  16-03-2020  Gov need to do somethings instead of biar je r...   
3797  16-03-2020  I and @ForestandPaper members are committed to...   

               Sentiment  
102             Positive  
103             Positive  
104              Neutral  
105   Extreme

Menggunakan metode .loc

In [20]:
# Menggunakan metode .loc (label indeks dan nama kolom)
subset_loc = dF.loc[102:, dF.columns[3]:]  # Mulai dari baris 102 dan dari kolom ke-4

print("\nMenggunakan metode .loc:")
print(subset_loc)


Menggunakan metode .loc:
         TweetAt                                      OriginalTweet  \
102   10-03-2020  Missouri AG sues televangelist for advertising...   
103   10-03-2020  President Trump: "It will go away.  Just stay ...   
104   10-03-2020  You actually need to stock pile on food first ...   
105   10-03-2020  We (@hamiltonpoverty) are calling on Canadian ...   
106   10-03-2020  Solid advice for seniors, their family members...   
...          ...                                                ...   
3793  16-03-2020  Meanwhile In A Supermarket in Israel -- People...   
3794  16-03-2020  Did you panic buy a lot of non-perishable item...   
3795  16-03-2020  Asst Prof of Economics @cconces was on @NBCPhi...   
3796  16-03-2020  Gov need to do somethings instead of biar je r...   
3797  16-03-2020  I and @ForestandPaper members are committed to...   

               Sentiment  
102             Positive  
103             Positive  
104              Neutral  
105   Extreme

Menampilkan jumlah label dari semua attribute yang bernilai object

In [18]:
# Mendapatkan attribute-attribute dengan tipe data 'object'
attribute_objek = dF.select_dtypes(include=['object'])
for attribute in attribute_objek.columns:
    jumlah_label = dF[attribute].nunique()
    print(f"Attribute '{attribute}' memiliki {jumlah_label} label.")

Attribute 'Location' memiliki 1717 label.
Attribute 'TweetAt' memiliki 15 label.
Attribute 'OriginalTweet' memiliki 3798 label.
Attribute 'Sentiment' memiliki 5 label.
